syntax = "proto3";

package api;

service Logger {
  rpc AppendMessage (AppendMessageRequest) returns (AppendMessageResponse) {}
  rpc ListMessages (ListMessagesRequest) returns (ListMessagesResponse) {}
}

message LogMessage {
  string log = 1;
}

message AppendMessageRequest {
  LogMessage log = 1;
}

message AppendMessageResponse {
}

message ListMessagesRequest {
}

message ListMessagesResponse {
  repeated LogMessage logs = 1;
}